### DISCUSSION TITLE
Message: [ Invalid routing data ]

### DISCUSSION THREAD

**Discussion by SF4User** [devemux86/cruiser#202] [Q&A] [answered]
Nice group tour today - 25 people. . .
The Scout navigated using Calimoto. . . everyone in our Group had the same fixed tour . . .
I have planned the Tour in the Kurviger III and I drive in the Cruiser.
The scout got lost four times  - I use cruiser and was always on the right route. That's how it should be 👍😉💪
Now to the problem:
We had to reschedule of course of a roadblock the tour . . . i.e. from the current position (A) directly to (B)
I enter in the cruiser navigation: our position = Start. . . new destination = Ziel
. . . Start navigation, message: [ Invalid routing data ]
So apparently I can't plan any routing's from the current position (A) directly to (B) in the cruiser.
No problem . . . Tour (A) to (B) planned in the Kurviger III and brought over in the Cruiser. . . Start navigating. . . everything OK!
Made a small correction in the cruiser navigation: and set a additional via point between (A) to (B).
Start navigation, message: [ Invalid routing data ]
Now the question: it isn't generally possible to plan a tour or change a planned tour in the cruiser?
Thanks for Answer (low priority) 😉

**Comment by devemux86** [comment]
@SF4User what had you selected in "Settings | Routes | Service"?

With message "Invalid routing data", seems you had "GraphHopper (offline)"?
If you had GraphHopper, did you also download the routing data for your area?
(menu "Downloads | GraphHopper")

However, it is recommended to use instead the "**BRouter (offline)**", please see:
- https://github.com/devemux86/cruiser/discussions/137

You can plan any route, but you do not need to do it manually inside navigation with Cruiser.

In case of roadblocks, you can press the top-left navigation panel and use the "**Avoid**" button.
It will calculate **offline** a side route until next waypoint, leaving your curvy route **unchanged**.

**Reply by SF4User** [reply, marked as answer]
Yes! Settings | Routes | Service was set on. . .
Your suggestion now: after downloading the routing data for my area (GraphHopper) it works !
Perfect! I stay in the GraphHopper world. . . I'm only "play" with BRouter later, when everything works.
Thanks for Service 🥰

**Reply by devemux86** [reply]
@SF4User thanks for the report!

I will change the message to mention GraphHopper. :slightly_smiling_face:

**Reply by SF4User** [reply]
@devemux86 
I've been working with this solution now. . . works perfect 👍
However, "only" one routing file can be activated - i.e. "only" e.g. Germany (GraphHopper).
Is it technically possible - ideally to activate at least more, i.e. three country routing files, also simultaneously.
We often change here borders on a tour. . . then the modified  offline routing ends at the national border
Germany/Austria/Italy . . . that's very quickly around here . . . 😉 from South Bavarian.

**Reply by devemux86** [reply]
@SF4User 

Unfortunately it is not possible to use multiple GraphHopper regions,
like we can use multiple offline maps or poi.

BRouter (offline) is the best alternative for cross border offline routing. :slightly_smiling_face:

**Reply by SF4User** [reply]
@devemux86 
OK! thanks for the feedback. It's not a real problem. . . at most a question of the comfort zone.
I will only switch to BRouter once I have understood the Cruiser as a whole. . . otherwise I never know where I just messed something up.
GraphHopper works very easy . . . my intention is: never change a running System (Iron Windows Rules . . . 😂😂)

**Comment by devemux86** [comment]
@SF4User 

In case you want to explore BRouter (offline) routing, you can check these settings (+ speed) to use smaller side roads:

More work is planned in Cruiser for offline routing and options. :slightly_smiling_face: 

<img src="https://github.com/devemux86/cruiser/assets/3484020/153ae992-b337-46a4-8a43-6a3993f0799a" width="50%"/>
